internal/handlers/common/update.go

```diff
@@ -44,7 +46,7 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 						return false, err
 					}
 				}
-				return true, nil
+				changed = true
 			default:
 				msgFmt := fmt.Sprintf(`Modifiers operate on fields but we found type %[1]s instead. `+
```

Comment:
We have the same error message in several places, including getUpdateOperatorDocument, which seems to be a common function that checks this case. Do we still need that check there?

Reply:
This exact error is never reached if that check is present.
should we panic here?

Reply:
In that case, we can drop the whole type switch and replace it with `setDoc := updateV.(*types.Document)` that would panic